### PR TITLE
Start using GC Flag TileEntities

### PR DESCRIPTION
**Pull request by Robotia**
TL;DR Fixes problem where this.loadedTIleEntityList.removeAll() takes forever for large tile entity volumes.

I encountered this issue on my KCauldron server when using ZettaIndustries' BigBattery. When the battery is massive, it creates an unprecedented volume of tile entities than can cause removing unloaded TEs to hang and leave the server unresponsive. Does not happen for Procyon's official builds, but does happen on these Spanner builds. 

Now, utilizing the GC flag I implemented yesterday, the loadedTileEntityList will be pruned for TEs marked for removal (setGC()).

Just in case, the GC flag is reset after removal from the list. I do not know whether Forge re-uses already loaded Tiles, so this will prevent the possibility of TEs that should be loaded not being ticked.

### CODE REVIEW

**Comment by Robotia** [comment]
I have not encountered any issues using this version on my server. I will post here if/when I do.
